tutorials/persistance-comparison-series/BinaryWriterReader/binary_reader_writer.md

```diff
@@ -0,0 +1,426 @@
+
+When you open up a clean 3D template, all you need to do is choose `GameObject` -> `3D Object` -> `Capsule`.
+
+You can then add scripts to the capsule by activating it in the hierarchy and using `Add Component` in the inspector.
+
+<img src="images/03_add_script.jpg" alt="Capsule in scene" width="250"/>
+<img src="images/04_script_name.jpg" alt="Capsule in scene" width="250"/>
+
+The scripts we will add to this capsule showcasing the different methods will all have the same basic structure that can be found in `HitCountExample.cs`.
+
+```cs
+using UnityEngine;
+
+/// <summary>
+/// This script shows the basic structure of all other scripts.
+/// </summary>
+public class HitCountExample : MonoBehaviour
+{
+    // Keep count of the clicks.
+    [SerializeField] private int hitCount; // 1
+
+    private void Start() // 2
+    {
+        // Read the persisted data and set the initial hit count.
+        hitCount = 0; // 3
+    }
+
+    private void OnMouseDown() // 4
+    {
+        // Increment the hit count on each click and save the data.
+        hitCount++; // 5
+    }
+}
+```
+
+The first thing we need to add is a counter for the clicks on the capsule (1). Add a `[SerilizeField]` here so that you can observe it while clicking on the capsule in the Unity editor.
+
+Whenever the game starts (2), we want to read the current hit count from the persistence and initialize `hitCount` accordingly (3). This is done in the `Start()` method that is called whenever a scene is loaded for each game object this script is attached to.
+
+The second part to this is saving changes, which we want to do whenever we register a mouse click. The Unity message for this is `OnMouseDown()` (4). This method gets called every time the `GameObject` that this script is attached to is clicked (with a left mouse click). In this case, we increment the `hitCount` (5) which will eventually be saved by the various options shown in this tutorials series.
+
+## BinaryReader and BinaryWriter
+
+(See `BinaryReaderWriterExampleSimple.cs` in the repository for the finished version.)
+
+In the previous tutorial, we looked at `Files`. This is not the only way to work with data in files locally. Another option that .NET is offering us is the [`BinaryWriter`](https://docs.microsoft.com/en-us/dotnet/api/system.io.binarywriter?view=net-5.0) and [BinaryReader](https://docs.microsoft.com/en-us/dotnet/api/system.io.binaryreader?view=net-5.0).
+
+> The BinaryWriter class provides methods that simplify writing primitive data types to a stream. For example, you can use the Write method to write a Boolean value to the stream as a one-byte value. The class includes write methods that support different data types.
+
+Parts of this tutorial will look familiar if you have worked through the previous one. We will use `File` again here to create and open file streams which can then be used by the `BinaryWriter` to save data into those files.
```

Comment:
I know that there is the quote from the documentation, but I think it would make sense if you explain with a little bit more words why using BinaryWriter would be an advantage over simply using a file. 
I suppose it's mostly the fact that you don't need to worry about how the file is structured, but just the order in which stuff is serialized/deserialized. Maybe it's also the fact that this is more space efficient than converting everything to string? (Just guessing)

Reply:
How about the approach in ac15b8e9decad9f95666d5e37b811f651dc4a7dc?

Reply:
Definitely better. If that's the only diff, then I guess you are done. I can't think of more, anyway.